go/cmd/remotetool/main.go

```diff
@@ -145,9 +147,14 @@ func main() {
 		}
 
 	case uploadDir:
-		if err := c.UploadDirectory(ctx, getPathFlag()); err != nil {
+		dg, err := c.UploadDirectory(ctx, getPathFlag())
+		if err != nil {
 			log.Exitf("error uploading directory for path %s: %v", getPathFlag(), err)
 		}
+		if *jsonOutput {
+			d, _ := json.Marshal(map[string]string{"digest": dg.String()})
```

Comment:
How hard would it be to add the cache hit rate and more metadata? In isolated I had included the individual file sizes as a delta-coded integer list. It was really sweet to create an histogram while packing the data as efficiently as possible; https://chromium.googlesource.com/infra/luci/luci-go/+/c364d2ae0a1c94014db11ff581ae7c9fa2ba1e3e/common/isolated/large.go
That said, sums and percentiles are probably sufficient.